cloud/scope/machine.go

```diff
@@ -299,11 +329,12 @@ func (m *MachineScope) Close(ctx context.Context) error {
 // the value from AzureMachine takes precedence.
 func (m *MachineScope) AdditionalTags() infrav1.Tags {
 	tags := make(infrav1.Tags)
-
 	// Start with the cluster-wide tags...
 	tags.Merge(m.ClusterDescriber.AdditionalTags())
 	// ... and merge in the Machine's
 	tags.Merge(m.AzureMachine.Spec.AdditionalTags)
+	// Set the cloud provider tag
+	tags[infrav1.ClusterAzureCloudProviderTagKey(m.ClusterName())] = string(infrav1.ResourceLifecycleOwned)
```

Comment:
we expect this to be permissive about state changes (cloud provider tag)? when?

Reply:
what do you mean by permissive about state changes? This was previously in https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/827/files#diff-b2813f64759b38fd374267409eaaca65L109 just moved it inside the func

Reply:
I was just looking at https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/cf2473927944a9585de7ae7d80abb2ac3369739b/api/v1alpha3/tags.go#L85-L96

I was reading that as `ClusterAzureCloudProviderTagKey` was for things like e.g. the ingress load balancer. I wasn't sure if the cloud provider would edit the VMs here

Reply:
pretty minor issue, just my curiosity on the usage of these

Reply:
yeah I'm not really sure why/how that tag is used for VMs, just preserved the existing tags